.travis.yml

```diff
@@ -0,0 +1,69 @@
+#
+# Turn off email notifications entirely
+#
+notifications:
+  email: false
+
+language: generic
+
+#
+# Define the build matrix
+#
+matrix:
+  include:
+#    - services: docker
+#      env: OS_RELEASE=debian_unstable
+#      sudo: required
+#      script:
+#        - docker-compose -f scripts/docker/docker-compose.yml build $OS_RELEASE
+#        - docker-compose -f scripts/docker/docker-compose.yml run $OS_RELEASE /bin/bash /workspace/code/scripts/travis.sh
+    - os: osx
+      osx_image: xcode7.3
+    - os: osx
+      osx_image: xcode7
+    - os: osx
+      osx_image: beta-xcode6.1
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      brew update &&
+      brew tap caskroom/cask &&
+      brew outdated cmake || brew install cmake &&
+      brew outdated swig || brew install swig &&
+      brew outdated gobject-introspection || brew install gobject-introspection &&
+      brew outdated python3 || brew install python3 &&
+      brew outdated python || brew install python &&
+      brew outdated pygobject3 || brew install pygobject3 --with-python --with-python3 &&
+      brew outdated lua || brew install lua &&
+      brew outdated yajl || brew install yajl &&
+      brew outdated augeas || brew install augeas &&
+      brew install Caskroom/cask/java &&
+      brew outdated libffi || brew install libffi &&
+      libffi_ver=$(brew ls --versions libffi | awk '{print $2}') && PKG_CONFIG_PATH=/usr/local/Cellar/libffi/${libffi_ver}/lib/pkgconfig/ /usr/local/bin/luarocks install lgi &&
+      export JAVA_HOME=$(/usr/libexec/java_home -v 1.8) &&
+      java -version &&
+      echo $JAVA_HOME;
+      export PATH=/usr/local/bin:/usr/local/sbin:$PATH;
+    fi
+#
+# Source is checked out in $TRAVIS_BUILD_DIR
+# Build directory will be TRAVIS_BUILD_DIR/../build
+#
+before_script:
+  - cd $TRAVIS_BUILD_DIR/..
+  - mkdir build && cd build
+  - CMAKE_OPT=()
+  - echo $PATH
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      python2_ver=$(python2 -c 'import sys; print(".".join(map(str, sys.version_info[:2])))') &&
+      CMAKE_OPT+=("-DPYTHON2_INCLUDE_DIR:PATH=$(python2-config --prefix)/include/python${python2_ver}") &&
+      CMAKE_OPT+=("-DPYTHON2_LIBRARY:FILEPATH=$(python2-config --prefix)/lib/libpython${python2_ver}.dylib") &&
+      python3_ver=$(python3 -c 'import sys; print(".".join(map(str, sys.version_info[:2])))') &&
+      CMAKE_OPT+=("-DPYTHON_INCLUDE_DIR:PATH=$(python3-config --prefix)/include/python${python3_ver}m") &&
+      CMAKE_OPT+=("-DPYTHON_LIBRARY:FILEPATH=$(python3-config --prefix)/lib/libpython${python3_ver}.dylib");
```

Comment:
I really don't like putting platform specific paths into *our* build system. If cmake is unable to find it, try to fix this at cmakes side or leave it. Otherwise you'll have to constantly keep up with platform specifics of other people, which fills our build system with obscure paths nobody can verify or clean up. Heck, right now cmake doesn't even find python3 on debian jessie or wheezy. And nobody cares about that, unless you're building this on our own. And even if you build elektra on your own, you usually pull the dependencies and configure line from the package files of your distribution and be done.

Beside that nobody stops a user from downloading python and installing into different paths. It just happens that brew is the easiest way on OSX so most users are using this packager right now. Might be another packager and other paths a few years later.

So instead, I suggest we do the same thing like on Linux: You (or a OSX maintainer) just write a definition file for brew in which you can put the special brew specific paths.

I have the same opinion regarding the automatic jna jar download.